controllers/discovery/searchUser.js

```diff
@@ -18,14 +18,27 @@ const SearchUser = async (req, res) => {
     });
 
   try {
+    let user_topics = await sequelize.query(
+      `select tp.topic_id from topics as tp
+      left join user_topics as utp on tp.topic_id = utp.topic_id
+      where utp.user_id = :userId`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          userId
+        }
+      }
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The query to fetch user topics is not handling potential SQL injection. It's important to sanitize inputs in SQL queries to prevent SQL injection attacks. In this case, `userId` should be sanitized before being used in the SQL query.
```diff
- where utp.user_id = :userId`,
+ where utp.user_id = ?`,
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->